src/main/java/uk/ac/cam/cl/dtg/isaac/quiz/IsaacNumericValidator.java

```diff
@@ -531,14 +545,14 @@ private QuantityValidationResponse exactStringMatch(final IsaacNumericQuestion i
                 }
 
                 if (questionAnswerString.toString().trim().equals(userStringForComparison.toString().trim())) {
-                    Boolean unitFeedback = null;
+                    Boolean unitsCorrect = null;
                     if (isaacNumericQuestion.getRequireUnits()) {
-                        unitFeedback = quantityFromQuestion.getUnits().equals(answerFromUser.getUnits());
+                        unitsCorrect = wasACorrectAnswerWithUsersSelectedUnit || quantityFromQuestion.isCorrect();
```

Comment:
Why is this necessary? `wasACorrectAnswerWithUsersSelectedUnit` has already been OR-ed with `quantityFromQuestion.isCorrect()` above so this is always redundant.